app/src/main/java/org/mozilla/focus/tiles/CustomTilesAccessor.kt

```diff
@@ -31,15 +31,50 @@ data class HomeTile (
             jsonObject.getString(KEY_ID))
 }
 
-object HomeTilesManager {
+/**
+ * Static accessor of custom home tiles, that is backed by SharedPreferences.
+ *
+ * New sites are appended to the end of the list.
+ *
+ * This keeps a cached version of the custom home tiles that have been pinned,
+ * in order to be more performant when checking whether sites are pinned or not.
+ * In order to keep the cache consistent, should only be called from the UIThread.
+ */
+object CustomTilesAccessor {
+    private lateinit var customTilesCache: LinkedHashMap<String, JSONObject>
+
+    // Cache pinned sites for perf beacues we need to check pinned state for every page load
+    fun initWithCache(customTiles: LinkedHashMap<String, JSONObject>) {
```

Comment:
nit: I think this should take a context and set the cache internally. The caller has no need to have access to the cache (and they could even mutate it if they wanted to).

Don't forget to privatize `getCustomTilesCache`.